### PR TITLE
IGNITE-16868 Fix build in IDEA

### PR DESCRIPTION
**Pull request by SammyVimes**
https://issues.apache.org/jira/browse/IGNITE-16868

### CODE REVIEW

File: modules/raft/pom.xml [hunk 1]
```diff
@@ -189,6 +158,20 @@
...
+                <configuration>
+                    <annotationProcessorPaths>
+                        <path>
+                            <groupId>org.apache.ignite</groupId>
+                            <artifactId>ignite-network-annotation-processor</artifactId>
+                            <version>${project.version}</version>
+                        </path>
+                        <path>
+                            <groupId>org.openjdk.jmh</groupId>
+                            <artifactId>jmh-generator-annprocess</artifactId>
```
Comment (sashapolo) [inline modules/raft/pom.xml:170]: why is this change needed? The previous configuration was used to avoid the `jmh` processor during non-test build

Comment (SammyVimes) [reply modules/raft/pom.xml:170]: IDEA doesn't know about different executions, so it doesn't execute annotation processor at all for raft module. It's also inconsistent with other modules, which use jmh.

**Review by sashapolo — APPROVED** [review]
